### DISCUSSION TITLE
Initial setup and configuration

### DISCUSSION THREAD

**Discussion by iivgp** [JanssenProject/jans#6313] [Q&A] [answered]
Hi, 

I am trying to get beyond initial setup for Janssen project.  I followed the installation guide for docker compose. The installation finished and the containers got up.

After logging into the jans container, I cannot get beyond authentication for TUI client. I am getting auth error message. 

Which client_id and credentials  should I use with the TUI? 
Where do I use admin password?

 I can access the healthcheck endpoint and the well-known endpoint. I could also connect to mysql and confirm that there is a set of clients provisioned. 

The curl command below results in authantication error as well:
curl -u "9876baac-de39-4c23-8a78-674b59df8c09:1t5Fin3#security"  https://demoexample.jans.io/jans-auth/restv1/token -k -d  "grant_type=client_credentials&scope=https://jans.io/oauth/config/scripts.write"

Can you point me in the right direction?

**Comment by iivgp** [comment]
If it helps, I do see /opt/jans/jans-setup/setup.properties.last file generated, but I do not see admin_password in it.

**Comment by nynymike** [comment]
@misba7 Can you take a look? Are the docs not clear how to launch the TUI after install?

**Comment by misba7** [comment]
Please follow this [doc](https://docs.jans.io/head/admin/kubernetes-ops/tui-k8s/) that helps you on how to install the tui, get the client id & secret and authenticate using them

**Comment by nynymike** [comment]
@misba7 is that page referenced on the monolith install page? And why don't we just install the TUI in the monolith itself? The goal of the monolith is convenience....

**Reply by misba7** [reply]
Yeah the tui is in the monolith container itself. Will also update the docs to make it more clear.

**Comment by misba7** [comment, marked as answer]
1. `docker exec -ti docker-jans-monolith-jans-1 bash`

2. `cat  setup.properties` and get the test_client_id and test_client_pw. These are the client id and client secret you will use with the config-cli/tui

3. Both config-cli and config-cli-tui come installed in the container and are located under `/opt/jans/jans-cli/`

4. So to use the tui for example: `python3 /opt/jans/jans-cli/config-cli-tui.py` and then pass the client id and secret you got in step 2

**Comment by iivgp** [comment]
Test client id and the password are default in this case. There were no configuration changes.

curl -u "9876baac-de39-4c23-8a78-674b59df8c09:1t5Fin3#security" https://demoexample.jans.io/jans-auth/restv1/token -k -d "grant_type=client_credentials"
{
    "error_description": "Client authentication failed (e.g. unknown client, no client authentication included, or unsupported authentication method). The authorization server MAY return an HTTP 401 (Unauthorized) status code to indicate which HTTP authentication schemes are supported. If the client attempted to authenticate via the Authorization request header field, the authorization server MUST respond with an HTTP 401 (Unauthorized) status code, and include the WWW-Authenticate response header field matching the authentication scheme used by the client. CorrelationId: 9d3f0087-a17f-4e63-8b16-989d4a2cfa7f",
    "error": "invalid_client"
}

**Reply by misba7** [reply]
Couldn't replicate that on my end.. It's working as expected.

![image](https://github.com/JanssenProject/jans/assets/49232835/811c9f6f-768c-426a-91c5-29b49f7c46ec)
